### PR TITLE
test(device): raise pods.go test coverage to 100% and add concurrency race test

### PR DESCRIPTION
**Pull request by Norway-02**
/kind test

## What this PR does / why we need it

This PR improves unit test coverage for `pkg/device/pods.go` (`PodManager`) from **84.8% to 100.0%** across all functions and statements.

`PodManager` serves as the core in-memory tracking engine for GPU resource allocations across nodes and containers. This PR adds test coverage for previously unverified methods and edge cases:

* **`DelPod`**: Verifies removal of existing pods and safe handling of non-existent pods (previously at 0% test coverage).
* **`ListPodsUID`**: Verifies UID slice generation across empty and populated pod maps (previously at 0% test coverage).
* **`AddPod`**: Tests the update branch when adding a pod that is already registered in cache.
* **`DeepCopy` Edge Cases**: Covers nil slice input behavior for `PodDevices`, `PodSingleDevice`, and `ContainerDevices`.
* **`TestPodManagerConcurrency`**: Introduces a multi-threaded stress test with 10 concurrent goroutines executing simultaneous reads, updates, additions, and deletions. This guarantees zero race conditions when verified under `go test -race`.

## Which issue(s) this PR fixes

N/A (test coverage improvement)

## Special notes for your reviewer

Strictly one test file (`pkg/device/pod_test.go`) is modified. No runtime code or GPU allocation logic is touched. Can be verified locally with:

```bash
go test -v -race -cover ./pkg/device/
```

## Does this PR introduce a user-facing change?

No.

---

**AI assistance disclosure:**

AI assistance was used to identify untested code paths and draft these unit tests. All tests were reviewed, run under the Go race detector, and verified manually. I take full responsibility for this contribution.

## Summary by CodeRabbit

* **Tests**
  * Added coverage for pod deletion and UID listing.
  * Added tests for updating existing pods.
  * Added edge-case coverage for nil deep copies.
  * Added concurrency tests for pod management operations.

### CODE REVIEW

**Comment by Norway-02** [comment]
@mesutoezdil @archlitchi @wawa0210 @DSFans2014 PTAL when you have a moment — this PR modifies strictly one test file (`pod_test.go`) to elevate `PodManager` statement coverage from 84.8% to 100.0% and adds a concurrency test under `go test -race`. Thank you!

File: pkg/device/pod_test.go [hunk 1]
```diff
@@ -555,3 +555,112 @@ func TestTakeAndDeletePodIsAtomic(t *testing.T) {
...
+	assert.True(t, ok)
+
+	// Delete existing pod
+	pm.DelPod(pod)
+	_, ok = pm.GetPod(pod)
+	assert.False(t, ok)
+
+	// Deleting a non-existing pod should not panic or cause errors
+	nonExistent := &corev1.Pod{ObjectMeta: metav1.ObjectMeta{UID: "uid-nonexistent"}}
+	pm.DelPod(nonExistent)
```
Comment (mesutoezdil) [inline pkg/device/pod_test.go:580]: nit: nothing is asserted after this, maybe check the pod count stayed the same so the no op is actually verified

**Review by mesutoezdil — COMMENTED** [review]
lgtm

**Comment by Norway-02** [comment]
@mesutoezdil Thank you for the LGTM and good suggestion! I've updated `TestDelPod` to add another pod before attempting the deletion, and added an explicit assertion verifying that attempting to delete a non-existent pod leaves the existing cache pod count unchanged.

**Comment by FouoF** [comment]
Thanks for adding PodManager tests. This is a coverage-only change with no linked defect or runtime behavior change, and the current branch does not pass the repository's basic lint gate (rangeint), causing the remaining checks to be skipped. The incremental coverage does not justify continuing review of a branch that is not green, so we are closing this PR. Please base future work on a maintainer-confirmed issue and ensure the full required checks run on the exact head.
